aptos-move/framework/src/module_metadata.rs

```diff
@@ -148,13 +152,21 @@ impl KnownAttribute {
     }
 }
 
+thread_local!(static METADATA_CACHE: RefCell<HashMap<Vec<Metadata>, Option<RuntimeModuleMetadataV1>>> = RefCell::new(HashMap::new()));
+
 /// Extract metadata from the VM, upgrading V0 to V1 representation as needed
 pub fn get_metadata(md: &[Metadata]) -> Option<RuntimeModuleMetadataV1> {
-    if let Some(data) = md.iter().find(|md| md.key == APTOS_METADATA_KEY_V1) {
+    if let Some(result) = METADATA_CACHE.with(|cache| cache.borrow_mut().get(md).cloned()) {
```

Comment:
I was hoping to get rid of this code if it didn't introduce more perf benefit. 

In terms of the choice of map, DashMap is definitely the concurrent hashmap that you requested in the other comment. Another option is LruCache so that we can always control the size of this cache. (also that was the reason why it was `borrow_mut()` because I thought that I will replace this hashmap with an LruCache so that insertion will require mutable reference than immutable ones.)

Reply:
Ping on this one.